samples/tools/testbed/README.md

```diff
@@ -0,0 +1,134 @@
+# Autogen Testbed Environment
+
+The Autogen Testbed environment is a tool for repeatedly running a set of pre-defined Autogen scenarios in a setting with tightly-controlled initial conditions. With each run, Autogen will start from a blank slate, working out what code needs to be written, and what libraries or dependencies to install. The results of each run are logged, and can be ingested by analysis or metrics scripts. By default, all runs are conducted in freshly-initialized docker containers, providing the recommended level of consistency and safety.
+
+## Setup
+
+Before you begin, you must configure your API keys for use with the Testbed. These keys extend beyond those typically found in an OAI_CONFIG_LIST, and can include such things as keys for the Bing Search API or other services used by the scenarios. There is an example ENV file in ``includes/ENV.example``. To get started:
```

Comment:
We need to improve consistency between setting up API keys between AutoGen core and samples

Reply:
Given the architecture, and the need to possibly pass multiple secrets to a Docker container (e.g., Bing Keys), I could not come up with more consistent way of handling API keys. All this does is source some environment variables, one of which is the OAI_CONFIG_LIST. These environment variables are exactly compatible with all other documentation. We *need* to get comfortable with scripts setting environment variables.

Reply:
Yes, the need for multiple secrets make sense. My comment was about the OAI_CONFIG_LIST, which is where we have common keys between AutoGen core and samples. Eg, one thing we could improve is that if a user has already setup OAI_CONFIG_LIST for autogen core -- either as an OAI_CONFIG_LIST env variable or file, this test suite should be able to reuse that?

Reply:
I can definitely check for an environment variable but there is no standard location for the OAI_CONFIG_LIST. It always looks in the cwd as far as I can tell. You would still need to copy the file from whatever other project you are working on to here.

Reply:
@gagb I added code here:

https://github.com/microsoft/autogen/blob/bfcf83e8173d869f0a96674c7ff17209afa9de8e/samples/tools/testbed/run_scenarios.py#L101

And here:

https://github.com/microsoft/autogen/blob/bfcf83e8173d869f0a96674c7ff17209afa9de8e/samples/tools/testbed/run_scenarios.py#L271

To use autogen's `config_list_from_json` to find and load the OAI_CONFIG_LIST. It will be consistent with this default behavior, or you can override with a new command-line option. 

The README has been updated as well. Please resolve this thread if it is satisfactory.

Reply:
Perfect this works!!